editor/store/selectors.js

```diff
@@ -1177,36 +1178,25 @@ export function getInserterItems( state, enabledBlockTypes = true ) {
 	return compact( items );
 }
 
-const getRecentInserts = createSelector(
-	state => {
-		// Filter out any inserts that are associated with a block type that isn't registered
-		const inserts = state.preferences.recentInserts.filter( insert => getBlockType( insert.name ) );
+function fillWithCommonBlocks( inserts ) {
+	// Filter out any inserts that are associated with a block type that isn't registered
+	const items = inserts.filter( insert => getBlockType( insert.name ) );
 
-		// Common blocks that we'll use to pad out our list
-		const commonInserts = getBlockTypes()
-			.filter( blockType => blockType.category === 'common' )
-			.map( blockType => ( { name: blockType.name } ) );
+	// Common blocks that we'll use to pad out our list
+	const commonInserts = getBlockTypes()
+		.filter( blockType => blockType.category === 'common' )
+		.map( blockType => ( { name: blockType.name } ) );
 
-		const areInsertsEqual = ( a, b ) => a.name === b.name && a.ref === b.ref;
-		return unionWith( inserts, commonInserts, areInsertsEqual );
-	},
-	state => state.preferences.recentInserts
-);
+	const areInsertsEqual = ( a, b ) => a.name === b.name && a.ref === b.ref;
+	return unionWith( items, commonInserts, areInsertsEqual );
+}
 
-/**
- * Determines the items that appear in the 'Recent' tab of the inserter.
- *
- * @param {Object}           state             Global application state.
- * @param {string[]|boolean} enabledBlockTypes Enabled block types, or true/false to enable/disable all types.
- *
- * @return {Editor.InserterItem[]} Items that appear in the 'Recent' tab.
- */
-export function getRecentInserterItems( state, enabledBlockTypes = true ) {
+function getItemsFromInserts( state, inserts, enabledBlockTypes = true, maximum = MAX_RECENT_BLOCKS ) {
```

Comment:
I'm wondering if we should we get rid of `MAX_RECENT_BLOCKS` and force callers to just always pass in a maximum.